code/_onclick/hud/screen_objects/text_objects.dm

```diff
@@ -8,3 +8,123 @@
 	screen_loc = "CENTER-7,CENTER-7"
 	maptext_height = 480
 	maptext_width = 480
+
+/// A screen object that shows the time left on a timer
+/atom/movable/screen/text/screen_timer
+	screen_loc = "CENTER-7,CENTER-7"
+	var/maptext_style_left = "<span class='maptext' style=font-size:16pt;text-align:center; align='top'>"
+	var/maptext_style_right = "</span>"
+	var/maptext_string
+	var/timer_id
+	var/list/timer_mobs = list()
+
+/atom/movable/screen/text/screen_timer/Initialize(
+		mapload,
+		mobs,
+		timer,
+		text,
+		style_start,
+		style_end,
+		offset_x = 12,
+		offset_y = -70
+	)
+	. = ..()
+
+	if(!islist(mobs))
+		if(!mobs)
+			return INITIALIZE_HINT_QDEL
+		mobs = list(mobs)
+	if(!timer)
+		stack_trace("Invalid timer for screen nuke timer!")
+		return INITIALIZE_HINT_QDEL
+	if(style_start)
+		maptext_style_left = style_start
+	if(style_end)
+		maptext_style_right = style_end
+	maptext_string = text
+	timer_id = timer
+	maptext_x = offset_x
+	maptext_y = offset_y
+	update_maptext()
+	if(length(mobs))
```

Comment:
Wont this always be true?

Reply:
Made it possible so that not having mobs is possible, thanks!